### PR TITLE
chore: add dynamic openapi joi example

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: examples/dynamic-openapi-joi.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const fastify = require('fastify')()
+const joiToJson = require('joi-to-json')
```
Comment (climba03003) [inline examples/dynamic-openapi-joi.js:4]:
```suggestion
// joi-to-json is one of the packages that can convert joi to json schema
// you are feel free to use other package that have the similar function
const convert = require('joi-to-json')
```

add a comment about `joi-to-json` can be changed to other packages that have similar ability.

File: examples/dynamic-openapi-joi.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  exposeRoute: true,
+  transform: schema => {
+    const {
+      params = undefined,
+      body = undefined,
+      querystring = undefined,
+      ...others
+    } = schema
+    const transformed = { ...others }
+    if (params) transformed.params = joiToJson(params)
```
Comment (climba03003) [inline examples/dynamic-openapi-joi.js:36]: use `convert` here to match with the docs.
```suggestion
    if (params) transformed.params = convert(params)
```

File: examples/dynamic-openapi-joi.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    if (body) transformed.body = joiToJson(body)
```
Comment (climba03003) [inline examples/dynamic-openapi-joi.js:37]: same reason
```suggestion
    if (body) transformed.body= convert(body)
```

File: examples/dynamic-openapi-joi.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    if (querystring) transformed.querystring = joiToJson(querystring)
```
Comment (climba03003) [inline examples/dynamic-openapi-joi.js:38]: same reason
```suggestion
    if (querystring) transformed.querystring = convert(querystring)
```

**Review by climba03003 — CHANGES_REQUESTED** [review]

**Review by climba03003 — APPROVED** [review]
LGTM

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**